cpp/src/gandiva/gdv_string_function_stubs.cc

```diff
@@ -264,6 +265,304 @@ const char* gdv_fn_lower_utf8(int64_t context, const char* data, int32_t data_le
   return out;
 }
 
+// converts utf8 to utf16 encoding
+GANDIVA_EXPORT
+const char* gdv_fn_encode(int64_t context, const char* data, int32_t data_len,
+                          const char* charset, int32_t charset_len, int32_t* out_len) {
+  if (data_len == 0) {
+    *out_len = 0;
+    return "";
+  }
+
+  char* out =
+      reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, 32 * data_len));
+  if (out == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
+    *out_len = 0;
+    return "";
+  }
+
+  int32_t char_len;
+  uint32_t char_codepoint;
+  int32_t y = 5;
+  int32_t z = 9;
+
+  std::string charset_str(charset, charset_len);
+
+  if (boost::iequals(charset_str, "UTF-16BE")) {
+    char* tmp = out;
+
+    int ind_w = 0;
+
+    for (int32_t i = 0; i < data_len; i += char_len) {
```

Comment:
You may want to use `arrow::util::UTF8ForEach` to trim this down severely...

Reply:
Well in all other functions such as lower and upper this was only used so I followed this approach.

Reply:
Well, that's a pity. In any case, it would be welcome to try relying on existing utilities.